allowances/contracts/AlowanceModule.sol

```diff
@@ -333,8 +338,15 @@ contract AllowanceModule is SignatureDecoder {
                 emit DeleteAllowance(msg.sender, delegate, token);
             }
         }
-        delegates[msg.sender][current.prev].next = current.next;
-        delegates[msg.sender][current.next].prev = current.prev;
+        if (current.prev == 0) {
```

Comment:
In `if(current.delegate == address(0)) return;`, shouldn't you `revert`? This would prevent users spending gas, as `call()` would look successful.

One space is missing also `if (current.delegate == address(0)) return;`

Reply:
you could revert ... you don't really safe gas (actually in case the tx is submitted the `revert` with a message, costs more gas). Only case would be that when you try to estimate it would show in some wallets.

Why I don't revert is also that you can do stuff like the following:
We have the setup with `Delegate A` and 1 safe owner proposes to `remove A and add B` and another propose to `remove A and add C` ... if both get approved you can in the current solution just execute both (as multisend tx), but if you revert, the one executed later will fail. 
Here the advantage is not that visible but in the `addDelegate` case this makes the client logic a lot easier as you can just always try to add a delegate, before setting an allowance (as just nothing will happen in case it already exists)

Reply:
Ok, understood :+1: